crates/wallet/src/wallet/mod.rs

```diff
@@ -533,168 +471,178 @@ impl Wallet {
         let mut indexed_graph = IndexedTxGraph::new(index);
         indexed_graph.apply_changeset(changeset.indexed_tx_graph);
 
-        let persist = Persist::new(db);
+        let staged = ChangeSet::default();
 
         Ok(Wallet {
             signers,
             change_signers,
             chain,
             indexed_graph,
-            persist,
+            staged,
             network,
             secp,
         })
     }
 
-    /// Either loads [`Wallet`] from persistence, or initializes it if it does not exist.
+    /// Either loads [`Wallet`] from the given [`ChangeSet`] or initializes it if one does not exist.
+    ///
+    /// This method will fail if the loaded [`ChangeSet`] has different parameters to those provided.
     ///
-    /// This method will fail if the loaded [`Wallet`] has different parameters to those provided.
+    /// ```rust,no_run
+    /// # use bdk_wallet::Wallet;
+    /// # use bdk_wallet::signer::{SignersContainer, SignerOrdering};
+    /// # use bdk_wallet::descriptor::Descriptor;
+    /// # use bitcoin::key::Secp256k1;
+    /// # use bdk_wallet::KeychainKind;
+    /// # use bdk_sqlite::{Store, rusqlite::Connection};
+    /// #
+    /// # fn main() -> Result<(), anyhow::Error> {
+    /// # use bitcoin::Network::Testnet;
+    /// # use bdk_chain::persist::Persist;
+    /// # let temp_dir = tempfile::tempdir().expect("must create tempdir");
+    /// # let file_path = temp_dir.path().join("store.db");
+    /// # let conn = Connection::open(file_path).expect("must open connection");
+    /// let mut db = Store::new(conn).expect("must create db");
+    /// let secp = Secp256k1::new();
+    ///
+    /// let external_descriptor = Descriptor::parse_descriptor(&secp, "wpkh(tprv8ZgxMBicQKsPdy6LMhUtFHAgpocR8GC6QmwMSFpZs7h6Eziw3SpThFfczTDh5rW2krkqffa11UpX3XkeTTB2FvzZKWXqPY54Y6Rq4AQ5R8L/84'/1'/0'/0/*)").unwrap();
+    /// let internal_descriptor = Descriptor::parse_descriptor(&secp, "wpkh(tprv8ZgxMBicQKsPdy6LMhUtFHAgpocR8GC6QmwMSFpZs7h6Eziw3SpThFfczTDh5rW2krkqffa11UpX3XkeTTB2FvzZKWXqPY54Y6Rq4AQ5R8L/84'/1'/0'/1/*)").unwrap();
+    ///
+    /// let changeset = db.load_changes()?;
+    /// let mut wallet = Wallet::new_or_load(external_descriptor, Some(internal_descriptor), changeset, Testnet)?;
+    /// # Ok(())
+    /// # }
+    /// ```
```

Comment:
```rust
/// ```rust,no_run
/// # use bdk_chain::persist::Persist;
/// # use bdk_wallet::Wallet;
/// # use bdk_sqlite::{Store, rusqlite::Connection};
/// # use bitcoin::Network::Testnet;
/// # let conn = Connection::open_in_memory().expect("must open connection");
/// let mut db = Store::new(conn).expect("must create db");
/// let changeset = db.load_changes()?;
///
/// let external_descriptor = "wpkh(tprv8ZgxMBicQKsPdy6LMhUtFHAgpocR8GC6QmwMSFpZs7h6Eziw3SpThFfczTDh5rW2krkqffa11UpX3XkeTTB2FvzZKWXqPY54Y6Rq4AQ5R8L/84'/1'/0'/0/*)";
/// let internal_descriptor = "wpkh(tprv8ZgxMBicQKsPdy6LMhUtFHAgpocR8GC6QmwMSFpZs7h6Eziw3SpThFfczTDh5rW2krkqffa11UpX3XkeTTB2FvzZKWXqPY54Y6Rq4AQ5R8L/84'/1'/0'/1/*)";
///
/// let mut wallet = Wallet::new_or_load(external_descriptor, Some(internal_descriptor), changeset, Testnet)?;
/// # Ok::<(), anyhow::Error>(())
/// ```    
```

Reply:
Thanks for making this example and the one below more succinct. I've updated as you suggest.